nuclear/cmake/CompilerWarnings.cmake

```diff
@@ -3,9 +3,9 @@
 # * https://github.com/lefticus/cppbestpractices/blob/master/02-Use_the_Tools_Available.md
 # * https://github.com/lefticus/cpp_starter_project/blob/master/cmake/CompilerWarnings.cmake
 
-function(set_project_warnings)
+function(set_target_warnings target_name)
```

Comment:
If you're going to move these into NUClear proper with some defaults then there needs to be an API to change them. NUClear roles is used in more than one project.
Have some configuration variable like the one that is used to set the banner/etc and then you can move these into the nuclear directory.